types/src/v2/params.rs

```diff
@@ -288,18 +288,49 @@ impl<'a> From<&'a [JsonValue]> for ParamsSer<'a> {
 #[derive(Debug, PartialEq, Clone, Hash, Eq, Deserialize, Serialize)]
 #[serde(deny_unknown_fields)]
 #[serde(untagged)]
-pub enum SubscriptionId {
+pub enum SubscriptionId<'a> {
 	/// Numeric id
 	Num(u64),
 	/// String id
-	Str(String),
+	#[serde(borrow)]
+	Str(Cow<'a, str>),
 }
 
-impl From<SubscriptionId> for JsonValue {
+impl<'a> From<SubscriptionId<'a>> for JsonValue {
 	fn from(sub_id: SubscriptionId) -> Self {
 		match sub_id {
 			SubscriptionId::Num(n) => n.into(),
-			SubscriptionId::Str(s) => s.into(),
+			SubscriptionId::Str(s) => s.into_owned().into(),
+		}
+	}
+}
+
+impl<'a> TryFrom<JsonValue> for SubscriptionId<'a> {
+	type Error = ();
```

Comment:
Would there be any use in making this more specific so that we can catch the "wrong value type" vs "value too large" errors below?

Reply:
yupp,